packages/opentelemetry-sdk-trace-base/test/common/export/BatchSpanProcessorBase.test.ts

```diff
@@ -175,43 +175,35 @@ describe('BatchSpanProcessorBase', () => {
       assert.strictEqual(spy.args.length, 0);
     });
 
-    it('should export the sampled spans with buffer size reached', done => {
-      const clock = sinon.useFakeTimers();
+    it('should export the sampled spans with buffer size reached', async () => {
       const processor = new BatchSpanProcessor(exporter, defaultBufferConfig);
-      for (let i = 0; i < defaultBufferConfig.maxExportBatchSize; i++) {
-        const span = createSampledSpan(`${name}_${i}`);
+      const span = createSampledSpan(name);
+      for (let i = 1; i < defaultBufferConfig.maxExportBatchSize; i++) {
         processor.onStart(span, ROOT_CONTEXT);
         assert.strictEqual(exporter.getFinishedSpans().length, 0);
 
         processor.onEnd(span);
         assert.strictEqual(exporter.getFinishedSpans().length, 0);
       }
-      const span = createSampledSpan(`${name}_6`);
       processor.onStart(span, ROOT_CONTEXT);
       processor.onEnd(span);
-
-      setTimeout(async () => {
-        assert.strictEqual(exporter.getFinishedSpans().length, 5);
-        await processor.shutdown();
-        assert.strictEqual(exporter.getFinishedSpans().length, 0);
-        done();
-      }, defaultBufferConfig.scheduledDelayMillis + 1000);
-      clock.tick(defaultBufferConfig.scheduledDelayMillis + 1000);
-      clock.restore();
+      assert.strictEqual(exporter.getFinishedSpans().length, 5);
+      await processor.shutdown();
+      assert.strictEqual(exporter.getFinishedSpans().length, 0);
     });
 
     it('should force flush when timeout exceeded', done => {
       const clock = sinon.useFakeTimers();
       const processor = new BatchSpanProcessor(exporter, defaultBufferConfig);
-      for (let i = 0; i < defaultBufferConfig.maxExportBatchSize; i++) {
-        const span = createSampledSpan(`${name}_${i}`);
+      const span = createSampledSpan(name);
+      for (let i = 1; i < defaultBufferConfig.maxExportBatchSize; i++) {
         processor.onStart(span, ROOT_CONTEXT);
         processor.onEnd(span);
         assert.strictEqual(exporter.getFinishedSpans().length, 0);
       }
 
       setTimeout(() => {
-        assert.strictEqual(exporter.getFinishedSpans().length, 5);
+        assert.strictEqual(exporter.getFinishedSpans().length, 4);
         done();
       }, defaultBufferConfig.scheduledDelayMillis + 1000);
```

Comment:
yes, previously the test did:

1. emit `maxExportBatchSize` spans
2. assert that `maxBatchSize` spans were exported after a timeout

This worked because we didn't eagerly export the batch once reaching the `maxExportBatchSize`

Therefore, the test has changed to emit `maxExportBatchSize - 1` spans and assert that that amount of spans was exported after the timeout